net.solarnetwork.node.io.mbus.jmbus/src/net/solarnetwork/node/io/mbus/jmbus/JMBusMBusNetwork.java

```diff
@@ -69,7 +74,9 @@ public MBusData read(int address) throws IOException {
 			return null;
 		}
 		conn.open();
-		return conn.read();
+		MBusData result = conn.read();
+		conn.close();
+		return result;
```

Comment:
Maybe this is clearer with a try-with-resource block like 
```java
try ( MBusConnection conn = createConnection(address) ) {
    conn.open();
    return conn.read();
}
```

Reply:
Oh yeah, that's much better. Thanks for bring that language feature to my attention :)
I've checked that change in.